### PR TITLE
[4.0] WebAsset for  JHtmlJquery,  JHtmlSearchtools

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

The patch add use of WebAsset for JHtmlJquery,  JHtmlSearchtools scripts.

### Testing Instructions

The site should work as before, without errors and visible changes.

_For reference #22435_

### CODE REVIEW

**Review by wilsonge — APPROVED** [review]

File: libraries/cms/html/jquery.php [hunk 1]
```diff
@@ -39,37 +39,26 @@
 	 * @return  void
 	 *
 	 * @since   3.0
+	 *
+	 * @deprecated 5.0  Use Factory::getDocument()->getWebAssetManager()->enableAsset('jquery');
```
Comment (laoneo) [inline libraries/cms/html/jquery.php]:
```suggestion
	 * @deprecated 5.0  Use $app->getDocument()->getWebAssetManager()->enableAsset('jquery');
```

File: libraries/cms/html/jquery.php [hunk 2]
```diff
@@ -39,37 +39,26 @@
...
-		}
-
-		// If no debugging value is set, use the configuration setting
-		if ($debug === null)
-		{
-			$debug = (boolean) Factory::getApplication()->get('debug');
-		}
-
-		HTMLHelper::_('script', 'vendor/jquery/jquery.min.js', array('version' => 'auto', 'relative' => true, 'detectDebug' => $debug));
+		$wa = Factory::getDocument()->getWebAssetManager();
```
Comment (laoneo) [inline libraries/cms/html/jquery.php]:
```suggestion
		$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
```

File: libraries/cms/html/jquery.php [hunk 3]
```diff
@@ -84,38 +73,21 @@ public static function framework($noConflict = true, $debug = null, $migrate = f
 	 * @return  void
 	 *
 	 * @since   3.0
+	 *
+	 * @deprecated 5.0  Use Factory::getDocument()->getWebAssetManager()->enableAsset('jquery.ui.core');
```
Comment (laoneo) [inline libraries/cms/html/jquery.php]:
```suggestion
	 * @deprecated 5.0  Use $app->getDocument()->getWebAssetManager()->enableAsset('jquery.ui.core');
```

File: libraries/cms/html/jquery.php [hunk 4]
```diff
@@ -84,38 +73,21 @@ public static function framework($noConflict = true, $debug = null, $migrate = f
...
 
-		// Include jQuery
-		static::framework();
-
-		// If no debugging value is set, use the configuration setting
-		if ($debug === null)
-		{
-			$debug = JDEBUG;
-		}
+		$wa = Factory::getDocument()->getWebAssetManager();
```
Comment (laoneo) [inline libraries/cms/html/jquery.php]:
```suggestion
		$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
```

File: libraries/cms/html/searchtools.php [hunk 5]
```diff
@@ -51,9 +51,7 @@ public static function form($selector = '.js-stools-form', $options = array())
 			$options = static::optionsToRegistry($options);
 
 			// Load the script && css files
-			HTMLHelper::_('behavior.core');
-			HTMLHelper::_('script', 'system/searchtools.min.js', array('version' => 'auto', 'relative' => true));
-			HTMLHelper::_('stylesheet', 'system/searchtools.css', array('version' => 'auto', 'relative' => true));
+			Factory::getDocument()->getWebAssetManager()->enableAsset('searchtools');
```
Comment (laoneo) [inline libraries/cms/html/searchtools.php]:
```suggestion
			Factory::getApplication()->getDocument()->getWebAssetManager()->enableAsset('searchtools');
```

**Review by laoneo — COMMENTED** [review]
Please use the document from the application as Factory::getDocument is deprecated.

**Comment by Fedik** [comment]
@laoneo updated, thanks!

**Comment by mbabker** [comment]
We shouldn't be using code snippets like this for deprecation statements.  Remember these render on docs pages like the API site.  So a `@deprecated 5.0  Use $app->getDocument()->getWebAssetManager()->enableAsset('jquery.ui.core');` tag has absolutely no context, WTF is `$app`?  A magic global?  A `stdClass` object?

If you're suggesting an alternative it should always be in the format of `Use <class>::<method>` instead.  Use the documentation for full code examples of how to use an API.

**Comment by Fedik** [comment]
@mbabker then `Factory::getApplication()->....` is fine, or?

**Comment by mbabker** [comment]
You know what, just use `Factory::getApplication()->whatever`.

The right way to do it would be to say `Use Joomla\CMS\WebAsset\WebAssetRegistry::enableAsset()` and leave it at that (because it's not a deprecation tag's or log message's responsibility to give anyone an exact copy/paste snippet on what the replacement is, the tag and notice are there to tell people what the replacement is and the core documentation would be the resource developers go to to learn how to actually use a feature/API, but we all know the docs are purely end user facing resources and the actual code has nothing but the automatically generated references from the doc blocks).  But people seem to think the doc blocks have to be self documenting on how to actually use the code, so whatever.

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
Sorry for the delay here. And thanks :)

**Pull request merged**